Solution1/Solution1.Tests/BL/Services/WeatherServiceTests.cs

```diff
@@ -119,5 +120,82 @@ public async Task GetForecastByCityNameAsync_ReturnedForecastWeatherDTO_Success(
 
             Assert.True(new CompareLogic().Compare(expectedWeatherDto, result).AreEqual);
         }
+
+        [Fact]
+        public async Task GetWeatherByArrayCityNameAsync_ReturnedDictionaryForWeatherResponseDTO_Success()
+        {
+            var cityName2 = "Paris";
+            var cityName3 = "AАА";
+            var cityName4 = string.Empty;
+            var weatherApiDto1 = new WeatherApiDTO() { CityName = cityName, TemperatureValues = new WeatherApiTempDTO() };
+            var weatherApiDto2 = new WeatherApiDTO() { CityName = cityName2, TemperatureValues = new WeatherApiTempDTO() };
+            
+            var listCityName = new List<string>() { cityName, cityName2, cityName3, cityName4 };
+            
+            var validationFailure = new List<ValidationFailure>() { new ValidationFailure("CityName", "Test validation error")};
+            var notValidResult = new ValidationResult(validationFailure);
+            var validResult = new ValidationResult(new List<ValidationFailure>());
+
+            SetWeatherApiServiceSettings(weatherApiDto1);
+            SetWeatherApiServiceSettings(weatherApiDto2);
+            SetWeatherApiServiceExceptionSettings(cityName3, new Exception());
+            SetWeatherApiServiceExceptionSettings(cityName4, new  ValidationException(validationFailure));
+
+            _validator
+                 .Setup(validator => validator.ValidateAsync
+                     (It.Is<ValidationContext<ForecastWeatherRequestDTO>>(context => context.InstanceToValidate.CityName == cityName4),
+                     It.IsAny<CancellationToken>()))
+                 .ReturnsAsync(notValidResult);
+            
+            _validator
+                 .Setup(validator => validator.ValidateAsync
+                     (It.Is<ValidationContext<ForecastWeatherRequestDTO>>(
+                         context => 
+                            context.InstanceToValidate.CityName == cityName
+                            || context.InstanceToValidate.CityName == cityName2
+                            || context.InstanceToValidate.CityName == cityName3),
+                     It.IsAny<CancellationToken>()))
+                 .ReturnsAsync(validResult);
+
+            // Act
+            var result = await _weatherService.GetWeatherByArrayCityNameAsync(listCityName);
+
+            // Assert
+            var isKeyTrue = result.Keys.Contains(true);
+            Assert.True(isKeyTrue);
+            if (isKeyTrue)
```

Comment:
if there isn't true key - first Assert will fail and will not continue test execution.
Use Contains instead true

Reply:
+